File: trash/src/count_monomers_by_chrom_and_len.py
import pandas as pd

d = pd.read_csv(
    "results/all_terminal_170_205.members.tsv",
    sep="\t"
)

counts = (
    d.groupby("chromosome")
     .size()
     .reset_index(name="n_monomers")
)

counts.to_csv(
    "results/terminal_170_205.counts_by_chromosome.tsv",
    sep="\t",
    index=False
)

d["length_class"] = pd.cut(
    d["width"],
    bins=[0, 169, 179, 189, 199, 205, 10**9],
    labels=["<170", "170-179", "180-189",
            "190-199", "200-205", ">205"]
)

(
    d.groupby(["chromosome", "length_class"], observed=True)
     .size()
     .reset_index(name="n_monomers")
     .to_csv(
         "results/terminal_monomers.by_chromosome_length.tsv",
         sep="\t",
         index=False
     )
)

File: trash/src/count_monomers_by_chrom_and_len.R
library(ggplot2)
library(readr)

d <- read_tsv(
  "results/terminal_monomers.by_chromosome_length.tsv"
)

plot <- ggplot(d, aes(chromosome, n_monomers, fill = length_class)) +
  geom_col() +
  labs(x = NULL, y = "Number of terminal monomers", fill = "Length") +
  theme_bw()

ggsave(plot, file = "results/terminal_monomers.by_chromosome_length.png")
